app/src/main/java/com/dsvag/currencyexchanger/ui/MainActivity.kt

```diff
@@ -41,11 +44,15 @@ class MainActivity : AppCompatActivity() {
     private fun apiCall() {
         repository.getCoins()
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe({
-                adapter.setData(it)
-            }, {
+            .subscribe({ adapter.setData(it) }, { })
+            .addTo(disposable)
+    }
 
-            })
+    private fun initSearchBar() {
+        binding.searchBar.textChanges()
+            .debounce(300, TimeUnit.MILLISECONDS)
+            .observeOn(AndroidSchedulers.mainThread())
```

Comment:
Да, я тупанул. Короче `debounce` по-умолчанию переключается на другой поток. Тут либо оставить твой `observeOn`, либо сказать, чтобы `debounce` работал на главном потоке (и мне кажется так лучше):
```suggestion
            .debounce(300, TimeUnit.MILLISECONDS, AndroidSchedulers.mainThread())
```

То же самое нужно сделать со вторым `debounce`-ом, который у тебя есть

Reply:
в адаптере который?